### PR TITLE
MetaDatApi Implementation

### PR DESCRIPTION
**Pull request by Sunaina04**

### CODE REVIEW

File: src/Type/ApiTypes.ts [hunk 1]
```diff
@@ -309,3 +321,17 @@ export interface WorkOrderResponse {
...
 }
+
+export interface MetaDataResponse {
+  message: string
+  status: number
+  errorList: []
+  time: number
+  content: {
+    data: {
+      name: string
```
Comment (ShKushal) [inline src/Type/ApiTypes.ts:332]: Please define it as a type or use existing if their is any

File: src/Components/CommonComponent/MetaDataComponent.tsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+} from '../../Services/MetaDataApi'
+import { MetaDataResponse } from '../../Type/ApiTypes'
+
+const useMetaData = () => {
+  const [getRoles] = useGetRolesMutation()
+  const [getCountries] = useGetCountriesMutation()
+  const [getStates] = useGetStatesMutation()
+
+  const fetchMetaData = async (getData: any) => {
+    try {
```
Comment (ShKushal) [inline src/Components/CommonComponent/MetaDataComponent.tsx:15]: for now onwards define these kind of API call functions in a separate file , 
increase re-usability

File: src/Components/AdminTools/AddNewCustomer.tsx [hunk 3]
```diff
@@ -39,24 +36,36 @@ const AddCustomer: React.FC<CustomerAdminDataProps> = ({ customerData, editMode
...
+      }
+      if (countriesData !== null) {
+        setCountriesData(countriesData)
+      }
+      if (statesData !== null) {
+        setStatesData(statesData)
+      }
+    }
+    fetchData()
+  }, [])
```
Comment (ShKushal) [inline src/Components/AdminTools/AddNewCustomer.tsx:67]: 1. don't add entire code in use effect, always create functions and call it in useEffect.
2. Memoize / useCallback on this function.
3. add a dependency to to useEffect (useCallback function)

File: src/Components/AdminTools/AddNewCustomer.tsx [hunk 4]
```diff
@@ -3,30 +3,27 @@ import { InputText } from 'primereact/inputtext'
...
-
-  const cities: CityProps[] = [
-    { name: 'New York', code: 'NY' },
-    { name: 'Rome', code: 'RM' },
-    { name: 'London', code: 'LDN' },
-    { name: 'Istanbul', code: 'IST' },
-    { name: 'Paris', code: 'PRS' },
-  ]
+  const { getMetaData } = useMetaData()
+  const [rolesData, setRolesData] = useState<Role[] | undefined>(undefined)
```
Comment (ShKushal) [inline src/Components/AdminTools/AddNewCustomer.tsx:24]: Redundant assignment .

In React ,  if you create a state variable using the useState hook but do not assign an initial value to it, the value of that state will be undefined by default.

File: src/Components/AdminTools/AddNewCustomer.tsx [hunk 5]
```diff
@@ -3,30 +3,27 @@ import { InputText } from 'primereact/inputtext'
...
   const [phone, setPhone] = useState('')
   const [email, setEmail] = useState('')
-  const [role, setRole] = useState<CityProps | undefined>(undefined)
   const [street, setStreet] = useState('')
   const [apt, setApt] = useState('')
-  const [state, setState] = useState<CityProps | undefined>(undefined)
-  const [country, setCountry] = useState<CityProps | undefined>(undefined)
   const [zipCode, setZipCode] = useState('')
+  const [role, setRole] = useState<Role | undefined>(undefined)
+  const [country, setCountry] = useState<Country | undefined>(undefined)
```
Comment (ShKushal) [inline src/Components/AdminTools/AddNewCustomer.tsx:19]: same here

**Review by ShKushal — CHANGES_REQUESTED** [review]

File: src/Components/CommonComponent/MetaDataComponent.tsx (hunk 2 above)
Comment (Sunaina04) [reply src/Components/CommonComponent/MetaDataComponent.tsx:15]: yes , this is the separate file actually for implementing metadata apis

File: src/Type/ApiTypes.ts [hunk 6]
```diff
@@ -322,7 +322,7 @@ export interface WorkOrderResponse {
   }
 }
 
-export interface MetaDataResponse {
+export type MetaDataResponse = {
```
Comment (ShKushal) [inline src/Type/ApiTypes.ts:325]: don't change interface to type.
This is what i meant.
image

**Review by ShKushal — APPROVED** [review]

**Pull request merged**